File: 2 Pointers/Two Sum II - Input Array Is Sorted/Solution.kt
class Solution {
    fun twoSum(numbers: IntArray, target: Int): IntArray {
        var leftPos = 0
        var rightPos = numbers.size - 1

        while (leftPos < rightPos){
            val sum = numbers[leftPos] + numbers[rightPos];
            if (sum == target){
                return intArrayOf(leftPos + 1 ,rightPos + 1)
            }else if (sum < target)
            {
                leftPos++
            }else
            {
                rightPos--
            }
        }
        return intArrayOf(0,0)
    }
}

File: 2 Pointers/Two Sum II - Input Array Is Sorted/Solution.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int leftPos = 0;
        int rightPos = numbers.size() - 1;

        while (leftPos < rightPos) {
            const int sum = numbers[leftPos] + numbers[rightPos];
            if (sum == target) {
                return {leftPos + 1, rightPos + 1};
            } else if (sum < target) {
                leftPos++;
            } else {
                rightPos--;
            }
        }
        return {0, 0};
    }
};